### ISSUE TITLE
Crash: App crashes when there is insufficient storage space on device to upload media

### ISSUE THREAD

**Issue by rachelmcr** [wordpress-mobile/WordPress-iOS#5043]
When a device is almost out of storage space, uploading large media files in the editor crashes the app.

### Expected behavior
If a device has insufficient storage space for the app to complete an action, I expect that the app would alert the user to the insufficient storage space on their device and prevent the action or otherwise handle the exception more gracefully.

### Actual behavior
If a device has insufficient storage space for the app to complete an action, for example when uploading media in the editor, the app crashes.

### Steps to reproduce the behavior
1. Fill your device's storage space until it is full/nearly full.
2. Open the WordPress app
3. Start a new post in the editor
4. Add a large media file (e.g. a video) from your device to the post

Result: The app crashes when it tries to upload the media file.

Crash log:
```
Exception Type:  SIGSEGV
Exception Codes: SEGV_ACCERR at 0xbbadbeef
Crashed Thread:  15
```

```
Thread 15 Crashed:
0   JavaScriptCore                       0x0000000184b9322c WTFCrash + 72
1   WebCore                              0x0000000184e848d8 _WebTryThreadLock(bool) + 264
2   WebCore                              0x0000000184e84f8c WebThreadLock + 104
3   UIKit                                0x0000000186685da0 -[UIThreadSafeNode respondsToSelector:] + 36
4   UIKit                                0x00000001862404a4 +[UIDictationController logCorrectionStatisticsForDelegate:] + 80
5   UIKit                                0x0000000186132e74 +[UIDictationController keyboardWillChangeFromDelegate:toDelegate:] + 296
6   UIKit                                0x00000001861321bc -[UIKeyboardImpl setDelegate:force:] + 444
7   UIKit                                0x000000018612beb0 -[UIPeripheralHost(UIKitInternal) _reloadInputViewsForResponder:] + 1176
8   UIKit                                0x000000018679d4bc -[UIPeripheralHost(UIKitInternal) _preserveInputViewsWithId:animated:reset:] + 484
9   UIKit                                0x0000000186485228 -[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1164
10  UIKit                                0x0000000186486d64 -[UIViewController _presentViewController:withAnimationController:completion:] + 4640
11  UIKit                                0x00000001864897a0 -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 468
12  UIKit                                0x00000001861ff2b4 -[UIViewController presentViewController:animated:completion:] + 180
13  WordPress                            0x0000000100253978 ext.WordPress.__ObjC.UIAlertController.presentFromRootViewController (__ObjC.UIAlertController)() -> () (UIAlertController+Helpers.swift:40)
14  WordPress                            0x0000000100253a38 @objc ext.WordPress.__ObjC.UIAlertController.presentFromRootViewController (__ObjC.UIAlertController)() -> () (UIAlertController+Helpers.swift:0)
15  WordPress                            0x00000001000e2ef4 +[WPError showAlertWithTitle:message:withSupportButton:okPressedBlock:] (WPError.m:170)
16  WordPress                            0x00000001000e2b68 +[WPError showAlertWithTitle:message:] (WPError.m:132)
17  WordPress                            0x0000000100043354 __44-[WPPostViewController addDeviceMediaAsset:]_block_invoke1062 (WPPostViewController.m:1804)
18  WordPress                            0x000000010035a950 partial apply forwarder for reabstraction thunk helper from @callee_unowned @convention(block) (@unowned __ObjC.NSError) -> (@unowned ()) to @callee_owned (@owned __ObjC.NSError) -> (@unowned ()) (PHAsset+Exporters.swift:0)
19  WordPress                            0x000000010035b674 partial apply forwarder for reabstraction thunk helper from @callee_unowned @convention(block) (@unowned __ObjC.NSError) -> (@unowned ()) to @callee_owned (@owned __ObjC.NSError) -> (@unowned ()) with unmangled suffix "14" (PHAsset+Exporters.swift:0)
20  WordPress                            0x000000010035d7dc ext.WordPress.__ObjC.PHAsset.(exportVideoToURL (__ObjC.PHAsset) -> (__ObjC.NSURL, targetUTI : Swift.String, maximumResolution : __C.CGSize, stripGeoLocation : Swift.Bool, successHandler : (resultingSize : __C.CGSize) -> (), errorHandler : (error : __ObjC.NSError) -> ()) -> ()).(closure #1).(closure #1) (PHAsset+Exporters.swift:144)
21  AVFoundation                         0x00000001876c7500 -[AVAssetExportSession(AVAssetExportSession_Local) _transitionToStatus:error:] + 472
22  libdispatch.dylib                    0x0000000180a054bc _dispatch_call_block_and_release + 20
23  libdispatch.dylib                    0x0000000180a0547c _dispatch_client_callout + 12
24  libdispatch.dylib                    0x0000000180a114c0 _dispatch_queue_drain + 860
25  libdispatch.dylib                    0x0000000180a08f80 _dispatch_queue_invoke + 460
26  libdispatch.dylib                    0x0000000180a13390 _dispatch_root_queue_drain + 724
27  libdispatch.dylib                    0x0000000180a130b0 _dispatch_worker_thread3 + 108
28  libsystem_pthread.dylib              0x0000000180c1d470 _pthread_wqthread + 1088
29  libsystem_pthread.dylib              0x0000000180c1d020 start_wqthread + 0
```

##### Tested on iPhone 6, iOS 9.3, app v. 6.0 and beta v. 6.1.0.20160318

**Comment by sendhil** [comment]
cc @SergioEstevao

**Comment by SergioEstevao** [comment]
This was fixed in the 6.1 release so I'm closing it.
